tests/unit/test_kraft.py

```diff
@@ -91,7 +91,7 @@ def test_ready_to_start_no_peer_cluster(charm_configuration, base_state: State):
 def test_ready_to_start_missing_data_as_controller(charm_configuration, base_state: State):
     # Given
     charm_configuration["options"]["roles"]["default"] = "controller"
-    charm_configuration["options"]["expose-external"]["default"] = "none"
+    charm_configuration["options"]["expose-external"]["default"] = "false"
```

Comment:
**praise**: Thanks for adding the validator in `structured_config.py` to spot this!

**suggestion, non-blocking**: since `false` is the default value, we could remove all those statements in this test and the next ones